### PR TITLE
[KYUUBI #5797][FOLLOWUP] Desc engine command support show engine registered attributes

### LINKED ISSUE #5797: [FEATURE] Support to describe engine and de-register engine with command in current session
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Support to describe the engine info of current session and de-register the engine.

For example:
```
KYUUBI DESC ENGINE;
KYUUBI DEREGISTER ENGINE;
```

### Motivation

Extensions for kyuubi sql commands.

### Describe the solution

Refer: https://github.com/apache/kyuubi/issues/3926

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by Kwafoor**
# :mag: Description

desc engine command support show engine registered attributes.

## Issue References 🔗

This pull request fixes #5797

## Describe Your Solution 🔧

https://github.com/apache/kyuubi/pull/5931#discussion_r1440679545

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/command/DescribeEngine.scala [hunk 1]
```diff
@@ -35,13 +35,19 @@ import org.apache.kyuubi.sql.schema.{Column, Row, Schema}
...
+      lazy val client = session.client
+      session.listZkEngineNodes.map { nodeInfo =>
+        val values = new ListBuffer[String]()
+        values += client.engineId.getOrElse("")
+        values += client.engineName.getOrElse("")
+        values += client.engineUrl.getOrElse("")
+        values += nodeInfo.namespace
+        values += s"${nodeInfo.host}:${nodeInfo.port}"
+        values += nodeInfo.version.getOrElse("")
+        values += nodeInfo.attributes.filter(_._1.contains("memory")).mkString(";")
```
Comment (Kwafoor) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/command/DescribeEngine.scala:48]: Is it necessary to filter Engine zk attributes? @turboFei

**Comment by Kwafoor** [comment]
image
`ENGINE_NAMESPACE` is a duplicate of `ENGINE_NAME`, delete  `ENGINE_NAMESPACE`

**Comment by Kwafoor** [comment]
@pan3793 @turboFei cc

File: kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/command/DescribeEngine.scala (hunk 1 above)
Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/command/DescribeEngine.scala:47]: not needed

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 2]
```diff
@@ -119,6 +120,12 @@ class KyuubiSessionImpl(
     engineLastAlive = System.currentTimeMillis()
   }
 
+  def listZkEngineNodes: Seq[ServiceNodeInfo] = {
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:123]: def getEngineNode: Option[ServiceNodeInfo]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala [hunk 3]
```diff
@@ -337,6 +337,15 @@ private[kyuubi] class EngineRef(
...
 
+  def getServiceNodes(
+      discoveryClient: DiscoveryClient,
+      hostPort: (String, Int)): Seq[ServiceNodeInfo] = {
+    tryWithLock(discoveryClient) {
+      val serviceNodes = discoveryClient.getServiceNodesInfo(engineSpace)
+      serviceNodes.filter { sn => (sn.host, sn.port) == hostPort }
+    }
+  }
+
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:348]: getServiceNode

Option[ServiceNodeInfo]

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:348]:
```
serviceNodes.filter { sn => (sn.host, sn.port) == hostPort }.headOption
```

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala [hunk 4]
```diff
@@ -42,6 +42,7 @@ import org.apache.kyuubi.util.{ThreadUtils, ThriftUtils}
 import org.apache.kyuubi.util.ThreadUtils.scheduleTolerableRunnableWithFixedDelay
 
 class KyuubiSyncThriftClient private (
+    hostPort: (String, Int),
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:45]: val hostPort: (String, Int),

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala [hunk 5]
```diff
@@ -52,6 +53,7 @@ class KyuubiSyncThriftClient private (
   @volatile private var _engineId: Option[String] = _
   @volatile private var _engineUrl: Option[String] = _
   @volatile private var _engineName: Option[String] = _
+  private[kyuubi] def engineHostPort: (String, Int) = hostPort
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:56]: not needed

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala [hunk 6]
```diff
@@ -52,6 +53,7 @@ class KyuubiSyncThriftClient private (
   @volatile private var _engineId: Option[String] = _
   @volatile private var _engineUrl: Option[String] = _
   @volatile private var _engineName: Option[String] = _
+  def engineHostPort: (String, Int) = hostPort
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:56]: you can remove this method I think, invoke the `hostPort` field directly.

**Review by turboFei — APPROVED** [review]
LGTM overall, left minor comments

**Comment by turboFei** [comment]
thanks @Kwafoor 
merged to 1.9

**Pull request closed without merging**